event/event.go

```diff
@@ -50,3 +51,69 @@ func (p *Publisher[T]) Publish(ctx context.Context, event T) error {
 		Body: encBody,
 	})
 }
+
+// RawSubscription represents a subscription that delivers messages as is.
+// No assumptions are made about the message contents. This should rarely be used in favor of [Subscription].
+type RawSubscription struct {
+	sub            *pubsub.Subscription
+	maxConcurrency int
+}
+
+// RawMessageHandler is responsible for handling raw messages from a subscription.
+type RawMessageHandler func([]byte) error
```

Comment:
Just worred if is this not too less? Don't we need to inject the log even for this case? I'm also afraid that this wouldn't allow us to add more things that may appear in the future, do you think using a struct perhaps would give more ways to extend?

In theory, our regular subscription will use the RawMessaHandler, right? Can we build everything on top of the []byte?

Reply:
Right now we can. I thought about this and was on the fence. We can't create a logger right now because there is no context that is useful to add on it, it would have the same effect as getting a default one. But I agree that it would make it more future proof..but at the same time this library is v0 and changing it should be fairly simple and then code using it would just need to do a small fix, so I started with the extreme minimum (but maybe I was too extreme ? :laughing: )

I'm OK with creating our own Message type and it would only have a Body for now. Maybe I was too Conan here.

![](https://github.com/katcipis/memes/blob/master/conan-approves.jpg?raw=true)

WDYT @lfritz ?

Reply:
Added a event.Message type for now, WDYT @guilherme-santos ?